x/batch/decorator_test.go

```diff
@@ -69,14 +95,15 @@ func mockTags(num int64) []common.KVPair {
 	return list
 }
 
-func mockData(num int64, content []byte) *batch.ByteArrayList {
+func mockData(num int64, content []byte) []byte {
 	list := &batch.ByteArrayList{}
 
 	for i := int64(0); i < num; i++ {
 		list.Elements = append(list.Elements, content)
 	}
 
-	return list
+	data, _ := list.Marshal()
```

Comment:
I think better take `testing.TB` as the first argument and check the error here. You can fail the test if the erorr is not nil.

Reply:
There is no need as it will fail down the line if something goes wrong here anyway. 
And it should never actually go wrong.